data/src/languages_update.py

```diff
@@ -0,0 +1,127 @@
+
+    Example: "ژۇرنال" -> ("Arabic", 1.0).
+    """
+    script_counts: Dict[str, float] = collections.defaultdict(float)
+    for char in word:
+        script_counts[unicodedataplus.script(char)] += 1.0
+    script_probs = [(s, script_counts[s] / len(word)) for s in script_counts]
+    script_probs.sort(key=operator.itemgetter(1), reverse=True)
+    if strict and len(script_probs) != 1:
+        return None
+    else:
+        # The script names in Unicode data tables have underscores instead of
+        # whitespace to enable parsing. See:
+        #   https://www.unicode.org/Public/13.0.0/ucd/Scripts.txt
+        return script_probs[0][0], script_probs[0][1]
+
+
+def _remove_duplicates(script_dict: Dict[str, str]) -> Dict[str, str]:
+    '''
+    If a values in lang["script"] appears more than once, the [key:value] pair that does not conform to ISO unicode
+    entries returned from unicodedataplus.property_value_aliases['script'] is deleted.
+
+    eg)
+
+    "aze": {
+        "iso639_name": "Azerbaijani",
+        "wiktionary_name": "Azerbaijani",
+        "wiktionary_code": "az",
+        "casefold": true,
+        "script": {
+            "latn": "Latin",
+            "cyrl": "Cyrillic",
+            "ara": "Arabic",
+            "arab": "Arabic"
+        }
+
+        ->
+
+        "aze": {
+        "iso639_name": "Azerbaijani",
+        "wiktionary_name": "Azerbaijani",
+        "wiktionary_code": "az",
+        "casefold": true,
+        "script": {
+            "latn": "Latin",
+            "cyrl": "Cyrillic",
+            "arab": "Arabic"
+        }
+
+
+    '''
+    remove = []
+
+    for key, value in script_dict["script"].items():
+        value = value.replace(" ", "_")
+        if not ''.join(unicodedataplus.property_value_aliases['script'][value]).lower() == key:
+            remove.append(key)
+    for i in remove:
+        del script_dict["script"][i]
+    return script_dict
+
+
+def _update_languages_json(tsv_path: str, output_path: str) -> None:
+    '''
+    Detects and identifies all unicode scripts present in a tsv file
+    and updates languages.json to reflect updated ["script"]
+    entries for each language in languages.json
+    '''
+    with open(LANGUAGES_PATH, "r", encoding="utf-8") as lang_source:
+        languages = json.load(lang_source)
+        for file in os.listdir(tsv_path):
+            if file.endswith('.tsv'):
+                iso639_code = file[:file.index("_")]
+                lang = languages[iso639_code]
+                with open(f'{tsv_path}/{file}', "r", encoding="utf-8") as f:
+                    for line in f:
+                        try:
+                            word = line.split("\t", 1)[0]
+                            script, prob = _detect_best_script_name(word)
+                            if not "script" in lang:
+                                lang["script"] = {}
+                            # use property_value_aliases to get ISO 15924 code
```

Comment:
`# Uses property_value_aliases to get ISO-15924 codes.`

Ideally these should look like real sentences.